wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/environment/server/compose/deserializer/EnvironmentDeserializer.java

```diff
@@ -0,0 +1,103 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.environment.server.compose.deserializer;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.google.common.collect.ImmutableMap;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.lang.String.format;
+
+/**
+ * Deserializer of environment section of Docker compose file.
+ *
+ * @author Dmytro Nochevnov
+ */
+public class EnvironmentDeserializer extends JsonDeserializer<Map<String, String>> {
+
+    public static final String ARRAY_ITEM_DIVIDER        = "=";
+    public static final String UNSUPPORTED_VALUE_MESSAGE = "Unsupported value '%s'.";
+
+    @Override
+    public Map<String, String> deserialize(JsonParser jsonParser, DeserializationContext ctxt) throws IOException, JsonProcessingException {
+        Object environment = jsonParser.readValueAs(Object.class);
+
+        /* Parse dictionary in view of:
+           "environment:
+             key1: value1
+             key2: value2"
+        */
+        if (environment instanceof Map) {
+            return (Map<String, String>) environment;
+        }
+
+        /* Parse array in view of:
+           "environment:
+             - key1=value1
+             - key2=value2"
+        */
+        try {
+            if (environment instanceof List) {
+                // convert array to Map<String, String>{ key1: value1, key2: value2 }
+                Map<String, String> map = new HashMap<>();
+                for (String item : (List<String>) environment) {
+                    map.putAll(parseEnvironmentItem(item));
+                }
+
+                return map;
+            }
+        } catch (IllegalArgumentException ie) {
+            throw ctxt.mappingException(ie.getMessage());
+
+        } catch (Exception e) {
+            throw ctxt.mappingException(format(UNSUPPORTED_VALUE_MESSAGE, environment));
+        }
+
+        // work around empty environment
+        if (environment instanceof String) {
+            if (((String) environment).isEmpty()) {
+                return new HashMap<>();
+            }
+
+            throw ctxt.mappingException(format(UNSUPPORTED_VALUE_MESSAGE, environment));
+        }
+
+        /* Work around unsupported type of environment content, for example, 'Boolean' in case of content:
+           "environment:
+             true"
+        */
+        throw ctxt.mappingException(format("Unsupported type '%s'.", environment.getClass()));
+    }
+
+    /**
+     * Parse "key=value" item.
+     * @param item - environment array item to parse
```

Comment:
Take a look good documented class here https://github.com/eclipse/che/blob/master/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/WorkspaceDao.java#L27
Also you can check your formatting by quick java doc feature. If you use it you'll see that there is double `-` after parameter